pkg/controller/daemon/util/daemon_util.go

```diff
@@ -147,6 +171,29 @@ func GetAllPodTemplates(daemon *extensions.DaemonSet, c clientset.Interface) (*a
 	return listPodTemplates(daemon, c)
 }
 
+// SetFromPodTemplate sets the desired PodTemplateSpec from a PodTemplate template to the given deployment.
+func SetFromPodTemplate(daemon *extensions.DaemonSet, podTemplate *api.PodTemplate) *extensions.DaemonSet {
+	daemon.Spec.Template.ObjectMeta = podTemplate.Template.ObjectMeta
+	daemon.Spec.Template.Spec = podTemplate.Template.Spec
+	daemon.Spec.Template.ObjectMeta.Labels = labelsutil.CloneAndRemoveLabel(
+		daemon.Spec.Template.ObjectMeta.Labels,
+		extensions.DefaultDaemonSetUniqueLabelKey)
+	return daemon
+}
+
+// SetiDaemonSetAnnotationsTo sets daemon set's annotations as given PodTemplate's annotations.
+// This action should be done if and only if the daemon set is rolling back to this PodTemplate.
+// Note that apply and revision annotations are not changed.
+func SetDaemonSetAnnotationsTo(daemon *extensions.DaemonSet, rollbackToPT *api.PodTemplate) {
+	// XXX:	deployment.Annotations = getSkippedAnnotations(deployment.Annotations)
+	daemon.Annotations = make(map[string]string)
+	for k, v := range rollbackToPT.Annotations {
+		// XXX:if !skipCopyAnnotation(k) {
```

Comment:
Add these back and have an empty map of the annotations we want to skip for now.